app/controllers/webhooks_controller.rb

```diff
@@ -120,55 +123,87 @@ def handle_payment_intent_succeeded(
 
       amount = item['amount']
       seller_id = item['seller_id']
-      merchant_name = Seller.find_by(seller_id: seller_id).name
-      case item['item_type']
-      when 'donation'
-        item = create_item(
-          item_type: :donation,
-          seller_id: seller_id,
-          purchaser: purchaser,
-          payment_intent: payment_intent
-        )
-        create_donation(item: item, amount: amount)
-        begin
-          send_donation_receipt(
-            payment_intent: payment_intent,
-            amount: amount,
-            merchant: merchant_name
-          )
-        rescue StandardError
+
+
+      if seller_id.eql?(POOL_DONATION_SELLER_ID)
+        unless item['item_type'].eql?('donation')
+          raise InvalidPoolDonationError,
+                "pool contribution must but be of type 'donation' but found type #{item['item_type']}."
         end
-      when 'gift_card'
-        purchaser = payment_intent.purchaser
-        item = create_item(
-          item_type: :gift_card,
-          seller_id: seller_id,
-          purchaser: purchaser,
-          payment_intent: payment_intent
-        )
 
-        recipient = payment_intent.recipient
-        gift_card_detail = create_gift_card(
-          item: item,
-          amount: amount,
-          seller_id: seller_id,
-          recipient: recipient
-        )
+        # calculate amount per merchant
+        # This will break if we ever have zero merchants but are still
+        # accepting pool donations.
+        amount_per = (amount.to_f/(Seller.count - 1).to_f).round
```

Comment:
Can you pull this into a separate method? It needs a bit more logic built in. It should round to the nearest cent since we can't have fractions of cents, then give the extra cents to each merchant in a random order (sort of like Venmo does when you use their divide calculator)

Reply:
I feel like we should else it'll be an accounting nightmare when cents don't add up and we have to reconcile which splits got rounded up vs which got rounded down. Sorry I know it's a pain! Slash a lot of engineering effort for something so dumb LOL